File: files-app/src/main/java/l/files/fs/FileTypeDetector.java
package l.files.fs;

import com.google.common.net.MediaType;

import java.io.IOException;

public interface FileTypeDetector {

    /**
     * Detects the content type of a file, if the file is a link returns the
     * content type of the target file.
     */
    MediaType detect(Path path) throws IOException;

    /**
     * Detects the content type of a file, use an existing status as hint.
     */
    MediaType detect(ResourceStatus status) throws IOException;

}

File: files-app/src/main/kotlin/fs/FileTypeDetector.kt
package l.files.fs

import com.google.common.net.MediaType

import java.io.IOException

trait FileTypeDetector {

    /**
     * Detects the content type of a file, if the file is a link returns the
     * content type of the target file.
     */
    throws(javaClass<IOException>())
    fun detect(path: Path): MediaType

    /**
     * Detects the content type of a file, use an existing status as hint.
     */
    throws(javaClass<IOException>())
    fun detect(status: ResourceStatus): MediaType

}
